### PR TITLE
Restyle "Buy ROSE" page and open Transak in new tab on small screens

### PR DESCRIPTION
**Pull request by lukaw3d**
If extension popup closes you lose all progress and it's too small to fit embedded widget.

| Before | After |
| --- | --- |
| wallet dev oasis io_account_oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km_fiat0 | localhost_3000_account_oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km_fiat |
| wallet dev oasis io_account_oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km_fiat1 | localhost_3000_account_oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km_fiat (1) |
| wallet dev oasis io_account_oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km_fiat(mocks1440x700)0 | localhost_3000_account_oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km_fiat (2) |
| wallet dev oasis io_account_oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km_fiat(mocks1440x700)1 | localhost_3000_account_oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km_fiat (3) |

### CODE REVIEW

File: src/app/pages/FiatOnrampPage/index.tsx [hunk 1]
```diff
@@ -14,27 +14,33 @@ import { networkActions } from '../../state/network'
...
+function Layout(props: { children?: React.ReactNode }) {
   const { t } = useTranslation()
   return (
     <Box
       round="5px"
       border={{ color: 'background-front-border', size: '1px' }}
       background="background-front"
       pad="medium"
+      alignSelf="center"
+      width="651px" // Padding + Transak threshold for >mobile layout
```
Comment (lubej) [inline src/app/pages/FiatOnrampPage/index.tsx:31]: `max-width` maybe? Kind of looks weird to me, setting a width. Not sure even extension has that much width available.

**Review by lubej — COMMENTED** [review]
Can confirm it works fine on mobile.

Not a fan that user's wallet address being sent via get parameters, even though it should be encrypted. Would be better if this was sent via a header or something. Just something to consider.

File: playwright/tests/extension.spec.ts [hunk 2]
```diff
@@ -43,7 +43,11 @@ test.describe('The extension popup should load', () => {
     await expect(popup.getByText('error').or(popup.getByText('fail'))).toBeHidden()
   })
 
-  test('should allow embedded Transak widget', async ({ page, extensionPopupURL }) => {
+  /**
+   * Skip test: Extension should still be able to show embedded Transak, but
+   * we link to it instead because closing the popup lost all progress.
+   */
+  test.skip('should allow embedded Transak widget', async ({ page, extensionPopupURL }) => {
```
[the remark this answers is not in the document]
Comment (lukaw3d) [reply playwright/tests/extension.spec.ts]: Re-enabled the test instead

File: src/app/pages/FiatOnrampPage/index.tsx [hunk 3]
```diff
@@ -47,56 +53,57 @@ export function FiatOnramp() {
   const isAddressInWallet = useSelector(selectIsAddressInWallet)
   const walletAddress = useSelector(selectAddress)
   const thirdPartyAcknowledged = useSelector(selectThirdPartyAcknowledged)
+  const isSmallScreen = useContext(ResponsiveContext) === 'small' || window.innerHeight < 700 // TODO: make reactive
```
[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/app/pages/FiatOnrampPage/index.tsx:56]: Thinking about this again, maybe we don't want it reactive. E.g. if you fill half of the form on a device in landscape mode, then rotate it, you should not lose your progress

Related https://github.com/oasisprotocol/oasis-wallet-web/issues/1714

[the remark this answers is not in the document]
Comment (lubej) [reply src/app/pages/FiatOnrampPage/index.tsx:56]: Oh, that sucks. But can we just always show the same fields, and not "toggle" anything with `if`s?

**Review by lubej — APPROVED** [review]

**Pull request merged**